xarray/core/dataarray.py

```diff
@@ -474,7 +474,10 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         if isinstance(key, basestring):
-            self.coords[key] = value
+            # convert to dataset first to use merge.dataset_update_method
```

Comment:
Can we fix this in `DataArray.coords.__setitem__` instead? Otherwise that will still be broken:
```
In [33]: da = xr.DataArray([0, 1, 2], dims='x')

In [34]: da.coords['x'] = [0, 1, 2, 3]
```

Reply:
You are right.
Updated. Thanks.